index.js

```diff
@@ -235,30 +238,12 @@ function processFormats(formats) {
   const filteredFormats = formats.filter(({ acodec, format_id, fps, height, protocol, vcodec }) => filterFormatCodecs(acodec, format_id, protocol, vcodec) && filterFormatFps(fps, height))
   .sort((prevFormat, nextFormat) => nextFormat.tbr - prevFormat.tbr);
 
+  // Select the best available option that has a max resolution of 4k, 1080p, and 720p
   const selected4K = filteredFormats.find((format) => (format.height === 2160 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 2160);
   const selected1080p = filteredFormats.find((format) => (format.height === 1080 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 1080);
-  const selectedLowQuality = filteredFormats.find((format) => (format.height === 720 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 720) || filteredFormats.find((format) => (format.height <= 720 && format.acodec !== 'none')) || filteredFormats.find((format) => (format.height <= 720));
-
-  const selectedFormats = [selected4K, selected1080p, selectedLowQuality].filter(Boolean).reduce((acc, format) => {
-    const type = format.fragments ? 'manifest' : 'url';
-    const assignedData = { type, ...format, url: format.url };
-
-    if (format.height === 2160) {
-      acc[`${format.acodec !== 'none' ? 'combined_4k' : '4k_video'}`] = assignedData;
-    }
-
-    if (format.height === 1080) {
-      acc[`${format.acodec !== 'none' ? 'combined_hd' : 'hd_video'}`] = assignedData;
-    }
-
-    if (format.height <= 720) {
-      acc[`${format.acodec !== 'none' ? 'combined_720p' : '720p_video'}`] = assignedData;
-    }
-
-    return acc;
-  }, {});
+  const selected720p = filteredFormats.find((format) => (format.height === 720 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 720) || filteredFormats.find((format) => (format.height <= 720 && format.acodec !== 'none')) || filteredFormats.find((format) => (format.height <= 720));
```

Comment:
Maybe `selectedLowQuality` is actually a better name here (since we aren't returning with the name anymore)

Reply:
fixed